tecken/settings.py

```diff
@@ -6,765 +6,787 @@
 Django settings for tecken project.
 """
 
+import json
 import logging
 import os
-from urllib.parse import urlparse
 
-from configurations import Configuration, values
+import dj_database_url
 from dockerflow.version import get_version
-import sentry_sdk
-from sentry_sdk.integrations.django import DjangoIntegration
-from sentry_sdk.integrations.logging import ignore_logger
+from everett.manager import ConfigManager, ListOf
 
 
-THIS_DIR = os.path.dirname(os.path.abspath(__file__))
-BASE_DIR = os.path.dirname(THIS_DIR)
-
-VERSION_FILE = get_version(BASE_DIR)
-
-SENTRY_DSN = os.environ.get("SENTRY_DSN", "")
-if SENTRY_DSN:
-    version = ""
-    if VERSION_FILE:
-        version = VERSION_FILE.get("version", "") or VERSION_FILE.get("commit", "")
-    version = version or "unknown"
-
-    sentry_sdk.init(
-        dsn=SENTRY_DSN,
-        release=version,
-        send_default_pii=False,
-        integrations=[DjangoIntegration()],
-    )
-    # Dockerflow logs all unhandled exceptions to request.summary so then Sentry reports
-    # it twice
-    ignore_logger("request.summary")
-    # This warning is unhelpful, so ignore it
-    ignore_logger("django.security.DisallowedHost")
+def json_parser(val):
+    try:
+        return json.loads(val)
+    except json.JsonDecodeError:
+        raise ValueError("not a valid JSON object")
 
-else:
-    print("SENTRY_DSN is not defined. SENTRY is not being set up.")
-
-
-DEFAULT_AUTO_FIELD = "django.db.models.AutoField"
 
+_config = ConfigManager.basic_config()
 
-class AWS:
-    "AWS configuration"
 
-    AWS_CONFIG = {
-        # AWS EC2 configuration
-        # 'AWS_REGION': 'us-west-2',
-        # 'EC2_KEY_NAME': '20161025-dataops-dev',
-    }
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+BASE_DIR = os.path.dirname(THIS_DIR)
 
+VERSION_FILE = get_version(BASE_DIR)
 
-# FIXME(willkg): 1728210: remove after we remove celery infra
-class Celery:
-
-    # Add a 5 minute soft timeout to all Celery tasks.
-    CELERY_TASK_SOFT_TIME_LIMIT = 60 * 5
-
-    # And a 10 minute hard timeout.
-    CELERY_TASK_TIME_LIMIT = CELERY_TASK_SOFT_TIME_LIMIT * 2
```

Comment:
I converted this file and then ran `manage.py diffsettings --all --output unified` on the main branch and this one and reconciled any differences of which there weren't many. That makes sure the settings were the same in the local dev environment between the main branch and this one.

We'll test server environments in stage.